aerofiles/flarmcfg/writer.py

```diff
@@ -210,3 +220,39 @@ def write_waypoints(self, points):
                 raise ValueError('Invalid number of task point tuple items')
 
             self.write_waypoint(*args)
+
+    def write_field(self, key, value):
+        if key == "pilot":
+            self.write_pilot(value)
+        elif key == "copilot":
+            self.write_copilot(value)
+        elif key == "competition_class":
+            self.write_competition_class(value)
+        elif key == "glider_type":
+            self.write_glider_type(value)
+        elif key == "glider_id":
+            self.write_glider_id(value)
+        elif key == "competition_id":
+            self.write_competition_id(value)
+        elif key == "logger_interval":
+            self.write_logger_interval(value)
+        elif key == "task_name":
+            self.write_task_declaration(value)
+        elif key == "waypoints":
+            self.write_waypoints(value)
+        else:
+            raise ValueError("There is no functions to write %s" % key)
+
+    def write_declared_task(self, task_dict):
```

Comment:
the main idea behind it was to reduce the number of lines in write_declared_task and reduce repetition.

the alternative could be the following (replacing the whole body of write_declared_task and yielding write_field obsolete). do you have a preference?

```
        if "pilot" in task_dict and task_dict["pilot"] is not None:
            self.write_pilot(task_dict["pilot"])

        if "copilot" in task_dict and task_dict["copilot"] is not None:
            self.write_copilot(task_dict["copilot"])

        if "competition_class" in task_dict and task_dict["competition_class"] is not None:
            self.write_competition_class(task_dict["competition_class"])

        if "glider_type" in task_dict and task_dict["glider_type"] is not None:
            self.write_glider_type(task_dict["glider_type"])

        if "glider_id" in task_dict and task_dict["glider_id"] is not None:
            self.write_glider_id(task_dict["glider_id"])

        if "competition_id" in task_dict and task_dict["competition_id"] is not None:
            self.write_competition_id(task_dict["competition_id"])

        if "logger_interval" in task_dict and task_dict["logger_interval"] is not None:
            self.write_logger_interval(task_dict["logger_interval"])

        if "task_name" in task_dict and task_dict["task_name"] is not None:
            self.write_task_declaration(task_dict["task_name"])
        else:
            raise TypeError("task_name should be contained in the task dictionary and can not be None")

        if "waypoints" in task_dict and task_dict["waypoints"] is not None:
            self.write_waypoints(task_dict["waypoints"])
        else:
            raise TypeError("waypoints should be contained in the task dictionary and can not be None")
```